### PR TITLE
Dala 2050 sme update

### PR DESCRIPTION
**Pull request by emanueldfine**
# Pull Request \

## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma

### CODE REVIEW

File: dataland-frontend/tests/e2e/fixtures/common/CountryFixtures.ts [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
- * Randomly returns a list of country codes
- * @returns the randomly generated list of country codes
+ * Randomly returns one Iso2 country code from all available Iso2 country codes
+ * @returns the randomly chosen country code
  */
-export function generateListOfIso2CountryCodes(): string[] {
-  return Array.from({ length: faker.number.int({ min: 1, max: 5 }) }, () => {
-    return generateIso2CountryCode();
-  }).sort((a, b) => a.localeCompare(b));
+export function getRandomIso2CountryCode(): string {
```
Comment (emanueldfine) [inline dataland-frontend/tests/e2e/fixtures/common/CountryFixtures.ts:9]: @fsoffel   faker has function to draw a random element from array

File: dataland-frontend/tests/e2e/fixtures/sme/SmePreparedFixtures.ts [hunk 2]
```diff
@@ -9,20 +9,18 @@ import { generateSmeFixtures } from "@e2e/fixtures/sme/SmeDataFixtures";
...
  * @param year the year as a number
  * @returns the manipulated fixture data
  */
-function manipulateFixtureForYear(input: FixtureData<SmeData>, year: number): FixtureData<SmeData> {
-  input.companyInformation.companyName = "SME-year-" + String(year);
-  input.t.financialYear = year;
-  input.reportingPeriod = String(year);
+function manipulateFixtureForYear(input: FixtureData<SmeData>, year: string): FixtureData<SmeData> {
+  input.companyInformation.companyName = "SME-year-" + year;
+  input.reportingPeriod = year;
```
Comment (emanueldfine) [inline dataland-frontend/tests/e2e/fixtures/sme/SmePreparedFixtures.ts:24]: @fsoffel  You can check if we even need this anymore

File: dataland-frontend/tests/e2e/fixtures/CompanyFixtures.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { faker } from "@faker-js/faker/locale/de";
```
Comment (stewal19) [inline dataland-frontend/tests/e2e/fixtures/CompanyFixtures.ts:1]: Is this only the german localization?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/Constants.kt [hunk 4]
```diff
@@ -41,6 +41,8 @@ val FRONTEND_DISPLAYED_FRAMEWORKS = listOf(
     DataTypeEnum.eutaxonomyMinusFinancials,
     DataTypeEnum.eutaxonomyMinusNonMinusFinancials,
     DataTypeEnum.lksg,
+    DataTypeEnum.sfdr,
+    DataTypeEnum.p2p,
```
Comment (stewal19) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/Constants.kt:45]: Why adding sfdr and p2p, but no sme?

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/utils/TestDataProvider.kt [hunk 5]
```diff
@@ -27,8 +26,8 @@ class TestDataProvider(@Autowired var objectMapper: ObjectMapper) {
             .map { it.copy(identifiers = IdentifierType.values().associateWith { emptyList() }) }
     }
 
-    fun getEmptySmeDataset(): SmeData {
```
Comment (stewal19) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/utils/TestDataProvider.kt:30]: Why was this changed?

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/controller/DataControllerTest.kt [hunk 6]
```diff
@@ -36,10 +36,10 @@ internal class DataControllerTest(
 ) {
     val testDataProvider = TestDataProvider(objectMapper)
 
-    val testDataType = DataType.valueOf("sme")
+    val testDataType = DataType.valueOf("eutaxonomy-non-financials")
     val storedCompanyEntity = testDataProvider.getEmptyStoredCompanyEntity()
-    val someSmeData = testDataProvider.getEmptySmeDataset()
```
Comment (stewal19) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/controller/DataControllerTest.kt:41]: Why those changes? Did this no longer work with sme data?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerInvestments.kt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package org.dataland.datalandbackend.model.sme.categories.power.subcategories
+
+import org.dataland.datalandbackend.model.enums.sme.PercentRangeForInvestmentsInEnergyEfficiency
+
+/**
+ * --- API model ---
+ * Fields of the subcategory "Investments" belonging to the category "Power" of the sme framework.
+*/
+data class SmePowerInvestments(
+    val percentageOfInvestmentsInEnhancingEnergyEfficiency: PercentRangeForInvestmentsInEnergyEfficiency? = null,
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerInvestments.kt:10]: Is this a good name for the variable type?

File: dataland-frontend/tests/e2e/fixtures/FixtureUtils.ts [hunk 8]
```diff
@@ -30,7 +31,29 @@ export function generateFixtureDataset<T>(
   return fixtureDataset;
 }
 
-export interface DataPoint<T, Y> {
-  label: string;
-  value: (x: T) => Y | undefined;
+/**
```
Comment (stewal19) [inline dataland-frontend/tests/e2e/fixtures/FixtureUtils.ts:34]: Can this changed sortly explained to me?

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/controller/DataControllerTest.kt (hunk 6 above)
Comment (fsoffel) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/controller/DataControllerTest.kt:41]: There was a required field added to SME and this method just returned an empty object. However, the type of dataset does not really matter here, thats why it was rather changed in data type.

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/Constants.kt (hunk 4 above)
Comment (fsoffel) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/Constants.kt:45]: I think only datasets with view pages are listed here.

File: dataland-frontend/tests/e2e/fixtures/CompanyFixtures.ts (hunk 3 above)
Comment (fsoffel) [reply dataland-frontend/tests/e2e/fixtures/CompanyFixtures.ts:1]: I changed many of the occurences of a german faker to the english one in a different branch already. However, through some merge conflict or something this one returned. And got fixed here again.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerInvestments.kt (hunk 7 above)
Comment (fsoffel) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerInvestments.kt:10]: We can discuss it. However, they are autogenerated through the new devtool outputs.

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/utils/TestDataProvider.kt (hunk 5 above)
Comment (fsoffel) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/utils/TestDataProvider.kt:30]: Because the function is not required anymore. Where it was used before, we can use a different data type. So there was no need for this one anymore.

Comment (stewal19) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/utils/TestDataProvider.kt:30]: ok

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/Constants.kt (hunk 4 above)
Comment (stewal19) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/Constants.kt:45]: ok makes sense

File: dataland-frontend/tests/e2e/fixtures/common/CountryFixtures.ts (hunk 1 above)
Comment (fsoffel) [reply dataland-frontend/tests/e2e/fixtures/common/CountryFixtures.ts:10]: Done.

File: dataland-frontend/tests/e2e/fixtures/sme/SmePreparedFixtures.ts (hunk 2 above)
Comment (fsoffel) [reply dataland-frontend/tests/e2e/fixtures/sme/SmePreparedFixtures.ts:24]: It gets used. However the year does not play a large role there. So 🤷 Keep it but remove the year?

File: dataland-frontend/tests/e2e/fixtures/FixtureUtils.ts (hunk 8 above)
Comment (stewal19) [reply dataland-frontend/tests/e2e/fixtures/FixtureUtils.ts:34]: Done in a call

File: dataland-frontend/tests/e2e/fixtures/lksg/LksgDataFixtures.ts [hunk 9]
```diff
@@ -47,46 +48,26 @@ export function generateOneLksgFixtureWithManyNulls(): FixtureData<LksgData> {
   )[0];
 }
```
Comment (ahoecherl) [inline dataland-frontend/tests/e2e/fixtures/lksg/LksgDataFixtures.ts:50]: Is there a good reason for these numerous changes to the Lksg Data Fixtures? Just cleaning up code? Addressing items from the refactoring log?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerConsumption.kt [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ * Fields of the subcategory "Consumption" belonging to the category "Power" of the sme framework.
+*/
+data class SmePowerConsumption(
+    val powerConsumptionInMwh: BigDecimal? = null,
+
+    val powerFromRenewableSources: YesNo? = null,
+
+    val energyConsumptionHeatingAndHotWater: BigDecimal? = null,
+
+    val primaryEnergySourceForHeatingAndHotWater: EnergySourceForHeatingAndHotWater? = null,
```
[the remark this answers is not in the document]
Comment (ahoecherl) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerConsumption.kt:19]: Field name directly adopted from data model. As the code is autogenerated we leave it as is.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerInvestments.kt (hunk 7 above)
Comment (ahoecherl) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/sme/categories/power/subcategories/SmePowerInvestments.kt:10]: As the name is aligned with the data dictionary category and sub-category and the naming of these categories is acceptable from my perspective leave it as is.
